app/src/main/java/org/permanent/permanent/ui/archiveOnboarding/compose/ArchiveOnboardingScreen.kt

```diff
@@ -185,20 +208,9 @@ data class NewArchive(
 )
 
 enum class OnboardingGoal {
```

Comment:
I feel you complicated a bit the logic on this page. I would do it like this:
1. Rename OnboardingGoal to OnboardingGoalType
enum class OnboardingGoalType {
CAPTURE, DIGITIZE, COLLABORATE, CREATE_AN_ARCHIVE, SHARE, CREATE_A_PLAN, ORGANIZE, SOMETHING_ELSE
}
2. Create a data class OnboardingGoal that stores the type, the text and if it's selected or not
data class OnboardingGoal(val type: OnboardingGoalType, val text: String, val checked: MutableState<Boolean>) 
3. In the ArchiveOnboardingScreen, instead of checkboxStates you create your goals like: 
val goals = rememberSaveable { mutableStateListOf(
        Goal(OnboardingGoal.CAPTURE, "Learn pro tips on how to capture and preserve.", mutableStateOf(false)),
        Goal(OnboardingGoal. DIGITIZE,"Digitize and transfer analog photos to a digital format.", mutableStateOf(false)),
      .....
    )}
4. Pass the goals to the GoalsPage instead of the: checkboxStates: CheckboxStates
fun GoalsPage(
    isTablet: Boolean, pagerState: PagerState, newArchive: NewArchive
    isTablet: Boolean,
    horizontalPaddingDp: Dp,
    pagerState: PagerState,
    newArchive: NewArchive,
    goals: OnboardingGoals
)
5. In the GoalsPage use a forEach to display the items instead of doing it one by one. If we have 50 items, it wouldn't be feasible to do this.
goals.forEach { goal ->
                CustomCheckbox(
                    isTablet = true,
                    text = goal.text,
                    checkedState = goal.checked
                )
        }